app/src/components/groups/SaveGroupVerificationCodeDialog.tsx

```diff
@@ -36,14 +36,25 @@ export function SaveGroupVerificationCodeDialog(props: SaveGroupVerificationCode
   const timeEllapsed = openedTimestamp ? Math.ceil((Date.now() - openedTimestamp) / 1000) : 0;
   const hasWaited = timeEllapsed >= MIN_WAIT_PERIOD_SECONDS;
 
+  const canClose = hasWaited && !isTransitioning;
+  function onClose() {
+    if (!canClose) {
+      return;
+    }
+
+    startTransition(() => {
+      props.onClose();
+    });
+  }
+
   return (
     <Dialog
       open={isOpen}
       onOpenChange={(val) => {
-        if (!val) props.onClose();
+        if (!val) onClose();
```

Comment:
On second thoughts and after seeing the changes in @rorro's proposed changes for this dialog, it might make sense to remove the Dialog's `onOpenChange` handler entirely... WDYT?